web-ts/src/routes/CityContentSwitcher.tsx

```diff
@@ -0,0 +1,118 @@
+  POIS_ROUTE,
+  LOCAL_NEWS_TYPE,
+  TU_NEWS_TYPE,
+  DISCLAIMER_ROUTE,
+  SEARCH_ROUTE,
+  CityModel,
+  useLoadFromEndpoint,
+  createLanguagesEndpoint,
+  LanguageModel,
+  CATEGORIES_ROUTE,
+  LANDING_ROUTE
+} from 'api-client'
+import { cmsApiBaseUrl } from '../constants/urls'
+import Layout from '../components/Layout'
+import FailureSwitcher from '../components/FailureSwitcher'
+import { LanguageFailure } from '../components/LanguageFailure'
+import GeneralHeader from '../components/GeneralHeader'
+import GeneralFooter from '../components/GeneralFooter'
+import LocationLayout from '../components/LocationLayout'
+import LoadingSpinner from '../components/LoadingSpinner'
+import { cityContentPattern, RoutePatterns } from './RootSwitcher'
+
+type PropsType = {
+  cities: CityModel[]
+} & RouteComponentProps<{ cityCode: string; languageCode: string }>
+
+// TODO pass right props instead of constants
+const CityContentSwitcher = ({ cities, match, location }: PropsType): ReactElement => {
+  const { cityCode, languageCode } = match.params
+  const cityModel = cities.find(it => it.code === cityCode)
+
+  const requestLanguages = useCallback(async () => {
+    return createLanguagesEndpoint(cmsApiBaseUrl).request({ city: cityCode })
+  }, [cityCode])
+  const { data: languages, loading, error: languagesError } = useLoadFromEndpoint<LanguageModel[]>(requestLanguages)
+  const languageModel = languages?.find(it => it.code === languageCode)
+
+  const landingPath = generatePath(RoutePatterns[LANDING_ROUTE], { languageCode })
+
+  if (!cityModel || !languageModel) {
+    if (loading) {
+      return (
+        <Layout
+          header={<GeneralHeader landingPath={landingPath} viewportSmall={false} />}
+          footer={<GeneralFooter language={languageCode} />}>
+          <LoadingSpinner />
+        </Layout>
+      )
+    }
+
+    const error = !cityModel ? new Error('notFound.category') : languagesError
+    if (error) {
+      return (
+        <Layout
+          header={<GeneralHeader landingPath={landingPath} viewportSmall={false} />}
+          footer={<GeneralFooter language={languageCode} />}>
+          <FailureSwitcher error={error} />
+        </Layout>
+      )
+    }
+
+    return (
+      <Layout
+        header={<GeneralHeader landingPath={landingPath} viewportSmall={false} />}
+        footer={<GeneralFooter language={languageCode} />}>
+        <LanguageFailure
+          cities={cities}
+          cityCode={cityCode}
+          pathname={location.pathname}
+          languageCode={languageCode}
+          languageChangePaths={[
+            { code: 'de', name: 'Deutsch', path: '/' },
+            { code: 'fr', name: 'French', path: '/' }
+          ]}
+          t={key => key}
+        />
+      </Layout>
+    )
+  }
+
+  return (
+    <LocationLayout
+      cities={cities}
+      categories={null}
+      viewportSmall={false}
+      feedbackTargetInformation={null}
+      languageChangePaths={null}
+      isLoading={false}
```

Comment:
Not sure about how we should handle this. Maybe keep a state here and pass callbacks to all routes to set it?

Reply:
To what do you refer here?

Reply:
Where this props come from and how to set/pass them. Of course becomes easy if we move the layout to the routes like you suggested.